src/main/resources/com/github/checkstyle/regression/extract/ExtractInfoGeneratorTest.java

```diff
@@ -40,12 +47,57 @@
  * @author LuoLiangchen
  */
 public class ExtractInfoGeneratorTest {
+    /** Modules which do not have global properties to drop. */
+    private static final List<String> XML_FILESET_LIST = Arrays.asList(
+            "TreeWalker",
+            "Checker",
+            "Header",
+            "Translation",
+            "SeverityMatchFilter",
+            "SuppressionFilter",
+            "SuppressWarningsFilter",
+            "BeforeExecutionExclusionFileFilter",
+            "RegexpHeader",
+            "RegexpOnFilename",
+            "RegexpSingleline",
+            "RegexpMultiline",
+            "JavadocPackage",
+            "NewlineAtEndOfFile",
+            "UniqueProperties",
+            "FileLength",
+            "FileTabCharacter"
+    );
+
+    /** Properties of abstract check. */
+    private static final Set<String> CHECK_PROPERTIES = getProperties(AbstractCheck.class);
+
+    /** Properties of abstract Javadoc check. */
+    private static final Set<String> JAVADOC_CHECK_PROPERTIES =
+            getProperties(AbstractJavadocCheck.class);
+
+    /** Properties of abstract file-set check. */
+    private static final Set<String> FILESET_PROPERTIES = getProperties(AbstractFileSetCheck.class);
+
+    /** Properties without document. */
+    private static final List<String> UNDOCUMENTED_PROPERTIES = Arrays.asList(
```

Comment:
We are not documenting properties here.
This is more a list of fields that aren't properties.